GoogleForCreators/web-stories-wp
# Editor: Hide upload buttons for users without upload capabilities

## Context



When clicking on select poster image no error message is shown and user is able select poster image to dismiss prepublish check list item. See video on #7755

## Summary

This PR hides all upload buttons and media editing capabilities if the user doesn't have upload capabilities (typically the "Contributor" user role in WordPress).

## Relevant Technical Choices



## To-do



## User-facing changes

| Before                | After                    |
| --------------- | --------------- |
| <img width="360" alt="Screenshot 2021-08-23 at 15 16 38" src="https://user-images.githubusercontent.com/237508/130463045-161c7c37-d6c1-4268-aa51-f070d52d1c4c.png"> | <img width="357" alt="Screenshot 2021-08-23 at 15 08 16" src="https://user-images.githubusercontent.com/237508/130462191-9fa06577-4d85-4848-a3b6-b63318286046.png"> | 
| <img width="302" alt="Screenshot 2021-08-23 at 15 16 54" src="https://user-images.githubusercontent.com/237508/130463208-08cda336-2fcb-430c-9e77-4fa9c015d122.png"> | <img width="300" alt="Screenshot 2021-08-23 at 15 09 26" src="https://user-images.githubusercontent.com/237508/130462309-55b4da95-0dd0-441f-8304-f6227b020a53.png"> | 
| <img width="295" alt="Screenshot 2021-08-23 at 15 17 00" src="https://user-images.githubusercontent.com/237508/130463342-b41ed10b-c461-4c65-b4ee-0a8fc7143af4.png"> | <img width="297" alt="Screenshot 2021-08-23 at 15 09 33" src="https://user-images.githubusercontent.com/237508/130462417-7ffcf779-5770-4b77-be2e-46c785450d64.png"> |
|   <img width="300" alt="Screenshot 2021-08-23 at 15 17 13" src="https://user-images.githubusercontent.com/237508/130463494-bdfa1153-4303-4d5e-8872-c0c78506bf91.png"> | <img width="305" alt="Screenshot 2021-08-23 at 15 09 47" src="https://user-images.githubusercontent.com/237508/130462531-301c0d94-263b-4cc4-a5e2-bf4adfce5ba3.png"> |
| <img width="303" alt="Screenshot 2021-08-23 at 15 20 44" src="https://user-images.githubusercontent.com/237508/130463613-fffddb8a-86a4-422a-aa7b-4bc6d536d813.png"> | <img width="297" alt="Screenshot 2021-08-23 at 15 10 08" src="https://user-images.githubusercontent.com/237508/130462587-786cd99f-fc5e-4ea2-8e37-6e230f8289c9.png"> |

## Testing Instructions



### QA






This PR can be tested by following these steps:

1. Login as contributor.
2. Open editor. 
3. Ensure that upload button does not appear in media library. 
4. Switch to documents tab. Edit icon for publisher logo and poster logo. 
5. Insert video into canva. 
6. Ensure that poster does not have edit icon. 
7. Ensure that there is no caption upload button appears. 
8. Add link to video. 
9. Ensure that there is no edit icon on link icon. 

### UAT







This PR can be tested by following these steps:

1.

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #7755
